tket-c-api/src/tket-c-api.cpp

```diff
@@ -0,0 +1,132 @@
+extern "C" {
+#include "tket-c-api.h"
+}
+
+#include <cstring>
+#include <system_error>
+
+#include "tket/Circuit/Circuit.hpp"
+#include "tket/Transformations/BasicOptimisation.hpp"
+#include "tket/Transformations/OptimisationPass.hpp"
+
+using namespace tket;
+using json = nlohmann::json;
+
+// Macro to suppress MSVC warning about throwing exceptions in extern "C"
+// functions
+#if defined(_MSC_VER)
+#define THROW_NO_MSVC_WARNING                                             \
```

Comment:
Hmm, MSVC does have a point: it is probably not a good idea use this as it is technically undefined behaviour. Either returning an error code or simply aborting with an assertion (if appropriate) would be better. That said, how do we get away without this warning at line 41?